sql: ALTER TYPE schema.GEOMETRY RENAME TO returns an internal error

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2937582&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2937582&tab=artifacts#/) on master @ [445175daed0848429219755703febfb3b50a73e9](https://github.com/cockroachdb/cockroach/commits/445175daed0848429219755703febfb3b50a73e9):


```
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1527
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1449
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:484
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:626
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | runtime.goexit
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +Wraps: (3) cannot use mutable descriptor of aliased type public.geometry
E210429 18:23:21.686087 2097149 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:46368,hostssl,user=root] 82 +Error types: (1) *assert.withAssertionFailure (2) *withstack.withStack (3) *errutil.leafError
    rsg_test.go:755: Crash detected: 
        ALTER TYPE ident . GEOMETRY RENAME TO ident
        
        Stack trace:
E210429 18:23:23.080863 2111037 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 83  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
    rsg_test.go:731: 4m0s of 5m0s: 863454 executions, 9583 successful
E210429 18:23:25.510393 2137196 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 84  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
E210429 18:23:27.191947 2154499 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 85  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
E210429 18:23:29.902748 2182974 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 86  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
    rsg_test.go:731: 4m5s of 5m0s: 884263 executions, 9818 successful
E210429 18:23:31.260741 2197594 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 87  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
E210429 18:23:32.656694 2212212 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 88  making txn commit explicit failed for "unnamed" meta={id=9c31abf4 pri=0.03934805 epo=0 ts=1619720612.654584159,0 min=1619720612.654584159,0 seq=2} lock=true stat=STAGING rts=1619720612.654584159,0 wto=false gul=1619720613.154584159,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
E210429 18:23:34.577982 2221501 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 89  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
    rsg_test.go:731: 4m10s of 5m0s: 897860 executions, 9982 successful
E210429 18:23:37.667792 2215516 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 90  making txn commit explicit failed for "unnamed" meta={id=107c8355 pri=0.01245965 epo=0 ts=1619720617.665436723,0 min=1619720617.665436723,0 seq=2} lock=true stat=STAGING rts=1619720617.665436723,0 wto=false gul=1619720618.165436723,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:731: 4m15s of 5m0s: 902260 executions, 10034 successful
    rsg_test.go:731: 4m20s of 5m0s: 913419 executions, 10162 successful
    rsg_test.go:731: 4m25s of 5m0s: 934368 executions, 10397 successful
E210429 18:23:53.674005 2349997 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 91  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
E210429 18:23:55.300198 2367781 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 92  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
    rsg_test.go:731: 4m30s of 5m0s: 956311 executions, 10677 successful
E210429 18:23:59.339430 2411081 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 93  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/195: cannot split range at key /Table/3/1/195
    rsg_test.go:731: 4m35s of 5m0s: 977895 executions, 10940 successful
E210429 18:24:00.636513 2425243 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 94  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
E210429 18:24:01.206941 2430937 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 95  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
E210429 18:24:04.033870 2461494 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 96  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
E210429 18:24:05.345617 2475196 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 97  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
    rsg_test.go:731: 4m40s of 5m0s: 999355 executions, 11201 successful
E210429 18:24:07.328118 2496933 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 98  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
E210429 18:24:09.609616 2520986 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 99  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/227: cannot split range at key /Table/3/1/227
E210429 18:24:10.256818 2528461 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 100  making txn commit explicit failed for "unnamed" meta={id=de4dbacd pri=0.00363393 epo=0 ts=1619720650.254798233,0 min=1619720650.254798233,0 seq=2} lock=true stat=STAGING rts=1619720650.254798233,0 wto=false gul=1619720650.754798233,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:731: 4m45s of 5m0s: 1021226 executions, 11459 successful
E210429 18:24:15.265703 2537253 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 101  making txn commit explicit failed for "unnamed" meta={id=0db4bbca pri=0.00006668 epo=0 ts=1619720655.263417223,0 min=1619720655.263417223,0 seq=2} lock=true stat=STAGING rts=1619720655.263417223,0 wto=false gul=1619720655.763417223,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:731: 4m50s of 5m0s: 1026150 executions, 11515 successful
    rsg_test.go:731: 4m55s of 5m0s: 1031791 executions, 11584 successful
    rsg_test.go:764: 1051730 executions, 11818 successful
    rsg_test.go:232: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration968719946
--- FAIL: TestRandomSyntaxGeneration (300.17s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #64009 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-queries @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944551&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944551&tab=artifacts#/) on master @ [684e753c15f3fc58df79b6ea70e7b6715eae4835](https://github.com/cockroachdb/cockroach/commits/684e753c15f3fc58df79b6ea70e7b6715eae4835):


```
        DROP TYPE FAMILY . GEOMETRY , FAMILY . INTERSECT . ident RESTRICT
        
        Stack trace:
E210501 18:22:24.655387 2159253 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 70  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:25.435312 2167814 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 71  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 3m50s of 5m0s: 849869 executions, 9961 successful
E210501 18:22:27.515073 2190535 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 72  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:29.792989 2215503 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 73  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 3m55s of 5m0s: 872119 executions, 10212 successful
E210501 18:22:31.561483 2236016 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 74  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:34.419517 2267602 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 75  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:35.562413 2280696 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 76  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m0s of 5m0s: 894482 executions, 10488 successful
E210501 18:22:37.845132 2306375 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 77  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:39.588387 2325394 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 78  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m5s of 5m0s: 916563 executions, 10770 successful
E210501 18:22:41.506785 2347459 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 79  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:44.342934 2378243 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 80  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m10s of 5m0s: 939054 executions, 11028 successful
E210501 18:22:45.677785 2393071 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 81  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:47.463086 2413274 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 82  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:49.479623 2435241 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 83  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m15s of 5m0s: 961043 executions, 11288 successful
E210501 18:22:51.581213 2458011 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 84  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:53.960479 2483920 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 85  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:55.478334 2500612 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 86  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m20s of 5m0s: 982870 executions, 11562 successful
E210501 18:22:57.605124 2523476 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 87  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:22:59.473307 2544637 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 88  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
    rsg_test.go:731: 4m25s of 5m0s: 1004945 executions, 11831 successful
E210501 18:23:01.570526 2567003 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 89  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:23:03.772741 2590514 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 90  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/233: cannot split range at key /Table/3/1/233
E210501 18:23:05.345731 2607477 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 91  making txn commit explicit failed for "unnamed" meta={id=54002463 pri=0.04105694 epo=0 ts=1619893385.343779799,0 min=1619893385.343779799,0 seq=2} lock=true stat=STAGING rts=1619893385.343779799,0 wto=false gul=1619893385.843779799,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
E210501 18:23:05.346810 2607322 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 92  making txn commit explicit failed for "unnamed" meta={id=144c293a pri=0.00242740 epo=0 ts=1619893385.345002575,0 min=1619893385.345002575,0 seq=2} lock=true stat=STAGING rts=1619893385.345002575,0 wto=false gul=1619893385.845002575,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:731: 4m30s of 5m0s: 1026385 executions, 12097 successful
    rsg_test.go:731: 4m35s of 5m0s: 1029212 executions, 12130 successful
    rsg_test.go:731: 4m40s of 5m0s: 1053699 executions, 12433 successful
    rsg_test.go:731: 4m45s of 5m0s: 1077453 executions, 12742 successful
E210501 18:23:21.640427 2751221 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 93  making txn commit explicit failed for "unnamed" meta={id=023ec771 pri=0.01938455 epo=0 ts=1619893401.637007282,0 min=1619893401.637007282,0 seq=2} lock=true stat=STAGING rts=1619893401.637007282,0 wto=false gul=1619893402.137007282,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
E210501 18:23:21.646148 2751268 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 94  making txn commit explicit failed for "unnamed" meta={id=227dc93f pri=0.00210190 epo=0 ts=1619893401.644034386,0 min=1619893401.644034386,0 seq=2} lock=true stat=STAGING rts=1619893401.644034386,0 wto=false gul=1619893402.144034386,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
E210501 18:23:22.453888 2758933 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 95  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/287: cannot split range at key /Table/3/1/287
    rsg_test.go:731: 4m50s of 5m0s: 1089249 executions, 12890 successful
    rsg_test.go:731: 4m55s of 5m0s: 1106755 executions, 13117 successful
E210501 18:23:34.730719 2857350 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 96  making txn commit explicit failed for "unnamed" meta={id=90c3e104 pri=0.02846043 epo=0 ts=1619893414.729297255,0 min=1619893414.729297255,0 seq=2} lock=true stat=STAGING rts=1619893414.729297255,0 wto=false gul=1619893415.229297255,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
E210501 18:23:34.747121 2857529 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 97  making txn commit explicit failed for "unnamed" meta={id=8d7e90e3 pri=0.01159362 epo=0 ts=1619893414.745327112,0 min=1619893414.745327112,0 seq=2} lock=true stat=STAGING rts=1619893414.745327112,0 wto=false gul=1619893415.245327112,0 ifw=1: EndTxn hasSystemKey=true, but hasSystemConfigTrigger=false
    rsg_test.go:731: 5m0s of 5m0s: 1128610 executions, 13368 successful
    rsg_test.go:764: 1128614 executions, 13369 successful
    rsg_test.go:232: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration111476312
--- FAIL: TestRandomSyntaxGeneration (304.56s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #64009 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-queries @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945604&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945604&tab=artifacts#/) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):


```
E210503 18:24:05.105253 2427735 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 97  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:07.288285 2449211 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 98  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m0s of 5m0s: 963569 executions, 11737 successful
E210503 18:24:09.547304 2471929 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 99  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:10.974273 2485825 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 100  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:13.247053 2509292 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 101  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m5s of 5m0s: 983129 executions, 11983 successful
E210503 18:24:15.076076 2526909 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 102  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:17.164482 2548342 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 103  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m10s of 5m0s: 1002970 executions, 12247 successful
E210503 18:24:19.507410 2571227 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 104  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:20.085207 2577408 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 105  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:21.742337 2593653 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 106  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:23.055575 2608913 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 107  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m15s of 5m0s: 1024007 executions, 12514 successful
E210503 18:24:25.374865 2633718 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 108  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:26.897787 2650268 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 109  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m20s of 5m0s: 1045946 executions, 12786 successful
E210503 18:24:29.133253 2675181 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 110  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:30.965050 2694826 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 111  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:33.061853 2715844 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 112  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:33.267616 2718284 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 113  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m25s of 5m0s: 1067197 executions, 13061 successful
E210503 18:24:35.317323 2740966 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 114  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:37.113927 2760469 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 115  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m30s of 5m0s: 1088965 executions, 13341 successful
E210503 18:24:39.888313 2790125 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 116  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:41.195365 2803971 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 117  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:42.933378 2822931 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 118  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m35s of 5m0s: 1110704 executions, 13585 successful
E210503 18:24:44.968820 2846143 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 119  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:47.878852 2877892 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 120  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:49.029794 2891780 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 121  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m40s of 5m0s: 1133114 executions, 13827 successful
E210503 18:24:51.256037 2915246 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 122  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:52.993468 2934410 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 123  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m45s of 5m0s: 1155246 executions, 14072 successful
E210503 18:24:55.164301 2958397 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 124  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:24:57.340501 2981702 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 125  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m50s of 5m0s: 1176687 executions, 14336 successful
E210503 18:24:59.774128 3008456 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 126  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:25:00.979323 3021690 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 127  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:731: 4m55s of 5m0s: 1198654 executions, 14569 successful
E210503 18:25:04.398741 3058825 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 128  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:25:05.019948 3064435 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 129  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
E210503 18:25:07.321584 3089130 kv/kvserver/queue.go:1093  [n1,split,s1,r6/1:/Table/{SystemCon…-11}] 130  unable to split [n1,s1,r6/1:/Table/{SystemCon…-11}] at key /Table/3/1/263: cannot split range at key /Table/3/1/263
    rsg_test.go:764: 1218726 executions, 14843 successful
    rsg_test.go:232: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration930605308
--- FAIL: TestRandomSyntaxGeneration (300.13s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #64009 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
- #63331 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-20.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-queries @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/descs/collection.go
+++ b/pkg/sql/catalog/descs/collection.go
@@ -551,18 +551,6 @@ func (tc *Collection) GetObjectDesc(
 	); isVirtual || err != nil {
 		return desc, err
 	}
-	// Resolve type aliases which are usually available in the PostgreSQL as an extension
-	// on the public schema.
-	// TODO(ajwerner): Pull this underneath type resolution.
-	if schema == tree.PublicSchema && flags.DesiredObjectKind == tree.TypeObject {
-		if alias, ok := types.PublicSchemaAliases[object]; ok {
-			if flags.RequireMutable {
-				return nil, errors.AssertionFailedf(
-					"cannot use mutable descriptor of aliased type %s.%s", schema, object)
-			}
-			return typedesc.MakeSimpleAlias(alias, keys.PublicSchemaID), nil
-		}
-	}
 	// Fall back to physical descriptor access.
 	switch flags.DesiredObjectKind {
 	case tree.TypeObject:

--- a/pkg/sql/logictest/testdata/logic_test/alter_type
+++ b/pkg/sql/logictest/testdata/logic_test/alter_type
@@ -619,3 +619,19 @@ ALTER TYPE typ_64101 DROP VALUE 'a'
 statement error could not remove enum value "b" as it is being used by table "test.public.t2_64101"
 ALTER TYPE typ_64101 DROP VALUE 'b'
 
+subtest regression_64398
+
+statement error pgcode 42809 type "geometry" is a built-in type
+ALTER TYPE sc64398.geometry RENAME TO bar
+
+statement error pgcode 42809 type "geometry" is a built-in type
+ALTER TYPE sc64398.public.geometry RENAME TO bar
+
+statement ok
+CREATE SCHEMA sc64398;
+
+statement ok
+CREATE TYPE sc64398.geometry AS ENUM()
+
+statement ok
+ALTER TYPE sc64398.geometry RENAME TO bar

--- a/pkg/sql/resolver.go
+++ b/pkg/sql/resolver.go
@@ -196,8 +196,19 @@ func (p *planner) LookupObject(
 	sc := p.Accessor()
 	lookupFlags.CommonLookupFlags.Required = false
 	lookupFlags.CommonLookupFlags.AvoidCached = p.avoidCachedDescriptors
-	objDesc, err := sc.GetObjectDesc(ctx, p.txn, dbName, scName, tbName, lookupFlags)
 
+	// Check if we are looking up a type which matches a built-in type in
+	// CockroachDB but is an extension type on the public schema in PostgreSQL.
+	if lookupFlags.DesiredObjectKind == tree.TypeObject && scName == tree.PublicSchema {
+		if alias, ok := types.PublicSchemaAliases[tbName]; ok {
+			if lookupFlags.RequireMutable {
+				return true, nil, pgerror.Newf(pgcode.WrongObjectType, "type %q is a built-in type", tbName)
+			}
+			return true, typedesc.MakeSimpleAlias(alias, keys.PublicSchemaID), nil
+		}
+	}
+
+	objDesc, err := sc.GetObjectDesc(ctx, p.txn, dbName, scName, tbName, lookupFlags)
 	return objDesc != nil, objDesc, err
 }
 

